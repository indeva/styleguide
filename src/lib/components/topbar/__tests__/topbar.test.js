import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Topbar from '..';

describe('Topbar', () => {
  const defaultProps = {
    avatarURL: 'https://placekitten.com/32/32',
    name: 'Felino da Silva',
    role: 'Gerente',
    links: [
      { title: 'Editar perfil', icon: ['far', 'pen'], url: '#1' },
      { title: 'Sair', icon: ['far', 'sign-out-alt'], url: '#2' },
    ],
    clickOnFilterButton: () => {},
    onLogoClick: () => {},
  };

  it('should renders properly', () => {
    const component = renderer.create(<Topbar {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('when there is children', () => {
    it('renders children with with-info modifier', () => {
      const component = renderer.create(
        <Topbar {...defaultProps}>
          <span>topbar</span>
        </Topbar>,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when state settingsActive is falsy (by default)', () => {
    it('should not include Settings', () => {
      const wrapper = mount(<Topbar {...defaultProps} />);
      expect(wrapper.exists('.topbar-settings')).toBeFalsy();
    });
  });

  describe('showFilterButton', () => {
    describe('when it is true', () => {
      it('should include Settings', () => {
        const component = renderer.create(<Topbar {...defaultProps} showFilterButton />);
        expect(component.toJSON()).toMatchSnapshot();
      });
    });

    describe('when it is false', () => {
      it('should include Settings', () => {
        const component = renderer.create(<Topbar {...defaultProps} showFilterButton={false} />);
        expect(component.toJSON()).toMatchSnapshot();
      });
    });
  });

  describe('when state settingsActive is truthly', () => {
    it('should include Settings', () => {
      const wrapper = mount(<Topbar {...defaultProps} />);
      wrapper.setState({ settingsActive: true });
      expect(wrapper.exists('.topbar-settings')).toBeTruthy();
    });
  });

  describe('.toggleSettings', () => {
    describe('when state settingsActive is falsy', () => {
      it('should change state to true', () => {
        const wrapper = mount(<Topbar {...defaultProps} />);
        wrapper.instance().toggleSettings();
        expect(wrapper.state('settingsActive')).toBeTruthy();
      });
    });

    describe('when state settingsActive is truthly', () => {
      it('should change state to false', () => {
        const wrapper = mount(<Topbar {...defaultProps} />);
        wrapper.setState({ settingsActive: true });
        wrapper.instance().toggleSettings();
        expect(wrapper.state('settingsActive')).toBeFalsy();
      });
    });
  });

  describe('on filter button click', () => {
    it('should trigger clickOnFilterButton prop fuction', () => {
      const spy = jest.spyOn(defaultProps, 'clickOnFilterButton');
      const wrapper = mount(<Topbar {...defaultProps} />);
      wrapper.find('.button').first().simulate('click');
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });

  describe('on logo click', () => {
    it('should trigger onLogoClick prop fuction', () => {
      const spy = jest.spyOn(defaultProps, 'onLogoClick');
      const wrapper = mount(<Topbar {...defaultProps} />);
      wrapper.find('.topbar__brand').first().simulate('click');
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});
