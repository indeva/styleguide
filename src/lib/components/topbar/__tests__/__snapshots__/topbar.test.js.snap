// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar should renders properly 1`] = `
<div
  className="topbar"
>
  <div />
  <div
    className="topbar__options"
  >
    <button
      className="button button--secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-filter fa-w-16 "
        data-icon="filter"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M463.952 0H48.057C5.419 0-16.094 51.731 14.116 81.941L176 243.882V416c0 15.108 7.113 29.335 19.2 40l64 47.066c31.273 21.855 76.8 1.538 76.8-38.4V243.882L497.893 81.941C528.042 51.792 506.675 0 463.952 0zM288 224v240l-64-48V224L48 48h416L288 224z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Filtros
    </button>
    <div
      className="topbar__settings"
    >
      <img
        alt="Felino da Silva"
        className="topbar__avatar"
        onClick={[Function]}
        role="presentation"
        src="https://placekitten.com/32/32"
      />
    </div>
  </div>
</div>
`;

exports[`Topbar showFilterButton when it is false should include Settings 1`] = `
<div
  className="topbar"
>
  <div />
  <div
    className="topbar__options"
  >
    <div
      className="topbar__settings"
    >
      <img
        alt="Felino da Silva"
        className="topbar__avatar"
        onClick={[Function]}
        role="presentation"
        src="https://placekitten.com/32/32"
      />
    </div>
  </div>
</div>
`;

exports[`Topbar showFilterButton when it is true should include Settings 1`] = `
<div
  className="topbar"
>
  <div />
  <div
    className="topbar__options"
  >
    <button
      className="button button--secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-filter fa-w-16 "
        data-icon="filter"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M463.952 0H48.057C5.419 0-16.094 51.731 14.116 81.941L176 243.882V416c0 15.108 7.113 29.335 19.2 40l64 47.066c31.273 21.855 76.8 1.538 76.8-38.4V243.882L497.893 81.941C528.042 51.792 506.675 0 463.952 0zM288 224v240l-64-48V224L48 48h416L288 224z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Filtros
    </button>
    <div
      className="topbar__settings"
    >
      <img
        alt="Felino da Silva"
        className="topbar__avatar"
        onClick={[Function]}
        role="presentation"
        src="https://placekitten.com/32/32"
      />
    </div>
  </div>
</div>
`;

exports[`Topbar when there is children renders children with with-info modifier 1`] = `
<div
  className="topbar topbar--with-info"
>
  <div>
    <span>
      topbar
    </span>
  </div>
  <div
    className="topbar__options"
  >
    <button
      className="button button--secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-filter fa-w-16 "
        data-icon="filter"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M463.952 0H48.057C5.419 0-16.094 51.731 14.116 81.941L176 243.882V416c0 15.108 7.113 29.335 19.2 40l64 47.066c31.273 21.855 76.8 1.538 76.8-38.4V243.882L497.893 81.941C528.042 51.792 506.675 0 463.952 0zM288 224v240l-64-48V224L48 48h416L288 224z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Filtros
    </button>
    <div
      className="topbar__settings"
    >
      <img
        alt="Felino da Silva"
        className="topbar__avatar"
        onClick={[Function]}
        role="presentation"
        src="https://placekitten.com/32/32"
      />
    </div>
  </div>
</div>
`;
